### ISSUE TITLE
Why doesn't the new version run smoothly？

### ISSUE THREAD

**Issue by eyinlijun** [GraphiteEditor/Graphite#804]
Hi, I am learning RUST and trying to implement some functions based on Graphite.
around the time the branche version with the color selection control becomes detached from the master version.
But when I recently tried to merge the mainline version, the imported SVG became awkward when zooming and moving.
Is it because the structure of the new version becomes complicated?

**Comment by Keavon** [comment]
One possibly: are you holding any keyboard keys down while zooming and moving?

**Comment by eyinlijun** [comment]
None of the things you mentioned happened.  
The intuitive thing is that when you have too many layers or anchors, the operation becomes very fluid.  
I have implemented SVG import here, and the difference will be obvious when importing the large image.  I can email you the code file (I have not considered submitting it for now because of the dependent license and code style issues).

**Comment by Keavon** [comment]
We do have some performance issues with the way that overlays are drawn, and lots of anchor points result in lots of overlays. Feel free to email me the patch set (or just post a link to your fork on GitHub) and I would be happy to take a look, thanks!

**Comment by 0HyperCube** [comment]
If you feel that this is a recent performance regression then it would be great if you could `git bisect` some of the recent changes to find the commit where this was introduced @eyinlijun.

**Comment by Keavon** [comment]
We have made some improvements in the past couple months to deal with these problems, like making the history system not store snapshots of the entire documents, including images, with unbounded growth. We have also filed issue #848 to deal with some of the other problems. As we move towards a fully node-driven architecture, we are hopeful that some of these performance issues will go away as we replace old or interim systems.
